pypechain/templates/contract.py/base.py.jinja2

```diff
@@ -48,7 +48,13 @@ from .{{struct_info.contract_name}}Types import {{struct_info.name}}
 {% for contract_type in link_references.contract_types %}
 from .{{contract_type}} import {{contract_type}}
 {% endfor %}
-from .utilities import dataclass_to_tuple, get_abi_input_types, rename_returned_types, tuple_to_dataclass
+
+{% if has_events %}
+from dataclasses import dataclass
+from pypechain.core import BaseEvent
+{% endif %}
+
+from pypechain.core import dataclass_to_tuple, get_abi_input_types, rename_returned_types, tuple_to_dataclass
```

Comment:
now that we are going to require pypechain in the generated package, should we consider also generating a `requirements.txt` or `pyproject.toml` that has the generating pypechain package version included as a requirement?